apps/mobile/src/components/widgets/tokens/tokens-widget.tsx

```diff
@@ -41,16 +44,84 @@ export function TokensWidget({ tokens, totalBalance }: TokensWidgetProps) {
           chain,
           fiatBalance,
         }) => (
-          <Flag key={ticker} img={icon} align="middle" spacing="1" reverse={false}>
-            <ItemLayout
-              titleLeft={tokenName}
-              titleRight={availableBalance && <Balance balance={availableBalance} />}
-              captionLeft={showChain(chain)}
-              captionRight={<Balance balance={fiatBalance} color="ink.text-subdued" />}
-            />
-          </Flag>
+          <TokenBalance
+            key={ticker}
+            availableBalance={availableBalance}
+            icon={icon}
+            ticker={ticker}
+            tokenName={tokenName}
+            chain={chain}
+            fiatBalance={fiatBalance}
+            showChain={showChain}
+          />
         )
       )}
-    </Widget>
+    </>
+  );
+}
+
+export function AccountBalances({ fingerprint, accountIndex }: AccountId) {
+  return (
+    <>
+      <BitcoinBalanceByAccount fingerprint={fingerprint} accountIndex={accountIndex} />
+    </>
+  );
+}
+
+interface BitcoinTokenBalanceProps {
+  availableBalance: Money;
+  fiatBalance: Money;
+}
+export function BitcoinTokenBalance({ availableBalance, fiatBalance }: BitcoinTokenBalanceProps) {
+  return (
+    <TokenBalance
+      ticker="BTC"
+      icon={<BtcAvatarIcon />}
+      tokenName={t`Bitcoin`}
+      chain={t`Bitcoin blockchain`}
+      fiatBalance={fiatBalance}
+      showChain={showChain}
+      availableBalance={availableBalance}
+    />
+  );
+}
+
+export function BitcoinBalance() {
```

Comment:
I see our preferred style here is to compose nested components for each token type that fetch their own data. 

I wasn't sure if it's better to fetch all the token balance data and cache it on the homepage first, as we will need that for the total balance anyway.

Maybe this is working the same way but on the `tokens-widget` page it should take the data from the cache

Reply:
> I see our preferred style here is to compose nested components for each token type that fetch their own data.

I'm not 100% on this, but the benefit is that it makes splitting requests easier. E.g. if this component is used in a single place, we can load bitcoin balances only. Think lazy load feature etc in future.

Happy for this to change, but think we should audit when we have both currencies in

Reply:
Thanks, I'm happy to go with this and we can assess if needed. It's a nice way of grouping things together more neatly too